src/audio/google/google_rtc_audio_processing.c

```diff
@@ -97,6 +108,130 @@ void GoogleRtcFree(void *ptr)
 	return rfree(ptr);
 }
 
+static ALWAYS_INLINE float clamp_rescale(float max_val, float x)
+{
+	float min = -1.0f;
+	float max = 1.0f - 1.0f / max_val;
+
+	return max_val * (x < min ? min : (x > max ? max : x));
+}
+
+static ALWAYS_INLINE float s16_to_float(const char *ptr)
```

Comment:
I feel like these should exist somewhere more generic so other components can use them if they are more optimized than the existing macros.